### PR TITLE
Layout update

### PR DESCRIPTION
**Pull request by daniel-martic-sociomantic**
For [New layout milestone](https://github.com/sociomantic-tsunami/nessie-ui/milestone/20?closed=1)

- `Grid` is now implemented with CSS `display: grid`
- implemented `GridItem` component that gives more control when using `Grid`
- removed built-in margins in components, most notably `margin-bottom` from most `.default:last-child`

### CODE REVIEW

File: src/Grid/index.jsx [hunk 1]
```diff
@@ -36,138 +47,112 @@ export default class Grid extends React.Component
...
-        onClick       : PropTypes.func,
+        customRows    : PropTypes.string,
         /**
-         *  onMouseOut callback function:
-         *  ( e ) => { ... }
+         *  Sets width to 100%
          */
-        onMouseOut    : PropTypes.func,
+        hasFullWidth  : PropTypes.bool,
         /**
```
Comment (conor-cafferkey-sociomantic) [inline src/Grid/index.jsx]: I think `hasFullWidth` can be removed(?)

File: src/Grid/index.jsx [hunk 2]
```diff
@@ -36,138 +47,112 @@ export default class Grid extends React.Component
...
+         * Horizontal alignment of the grid items
          */
-        onMouseOver   : PropTypes.func,
+        justify       : PropTypes.oneOf( [
+            'left', 'center', 'right', 'stretch' ] ),
         /**
          *  Grid role
          */
-        role          : PropTypes.string,
+        role   : PropTypes.string,
```
Comment (conor-cafferkey-sociomantic) [inline src/Grid/index.jsx]: `role` can be removed

File: src/Grid/index.jsx [hunk 3]
```diff
@@ -36,138 +47,112 @@ export default class Grid extends React.Component
...
-                    {
-                        console.warn( 'Grid / Row must have Columns as direct \
-children' );
-                        Grid.didWarn.children = true;
-                    }
-                } );
-            }
-        }
+        const layout = {
+            'gridAutoColumns' : autoCols !== undefined ?
```
Comment (conor-cafferkey-sociomantic) [inline src/Grid/index.jsx]: Quotes are not required around the object keys here. Also we probably don't need to explicitly check for `undefined`. I think we can just do: ` gridAutoColumns : autoCols || '1fr'`, for example.

File: src/NotificationBar/notificationBar.css [hunk 4]
```diff
@@ -19,28 +19,21 @@
...
-    }
-
     .message
     {
-        color: var( --PC-WHITE );
+        color:          var( --PC-WHITE );
     }
 
-    .icon
+    .info
```
Comment (conor-cafferkey-sociomantic) [inline src/NotificationBar/notificationBar.css]: This should still be `.icon`

File: src/Row/index.jsx [hunk 5]
```diff
@@ -7,70 +7,128 @@
...
+
+    static displayName = 'Row';
+
+    render()
+    {
+        const {
+            align,
+            children,
+            className,
+            cssMap = createCssMap( this.context.Radio, this.props ),
```
Comment (damian-rodriguez-sociomantic) [inline src/Row/index.jsx]: row context should be used instead of radio

File: src/GridItem/index.jsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+export default class GridItem extends React.Component
+{
+    static contextType = ThemeContext;
+
+    static propTypes =
+    {
+        /**
+         * Vertical alignment of the GridItem content
+         */
+        align     : PropTypes.oneOf( [ 'top', 'middle', 'bottom', 'stretch' ] ),
```
Comment (damian-rodriguez-sociomantic) [inline src/GridItem/index.jsx]: update proptypes according to the css classes

File: src/GridItem/index.jsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+         */
+        colSpan   : PropTypes.number,
+        /**
+         *  CSS class map
+         */
+        cssMap    : PropTypes.objectOf( PropTypes.string ),
+        /**
+         * Horizontal alignment of the GridItem content
+         */
+        justify   : PropTypes.oneOf( [ 'left', 'center', 'right', 'stretch' ] ),
```
Comment (damian-rodriguez-sociomantic) [inline src/GridItem/index.jsx]: update proptypes according to the css classes

**Pull request merged**